vcd/resource_vcd_org_vdc_access_control.go

```diff
@@ -0,0 +1,236 @@
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"shared_with_everyone": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Whether the VDC is shared with everyone",
+			},
+			"everyone_access_level": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{types.ControlAccessReadOnly}, true),
+				Description:  "Access level when the VDC is shared with everyone (only ReadOnly is available). Required when shared_with_everyone is set",
+			},
+			"shared_with": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"user_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "ID of the user to which we are sharing. Required if group_id is not set",
+						},
+						"group_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "ID of the group to which we are sharing. Required if user_id is not set",
+						},
+						"subject_name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Name of the subject (group or user) with which we are sharing",
+						},
+						"access_level": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{types.ControlAccessReadOnly}, true),
+							Description:  "The access level for the user or group to which we are sharing. (Only ReadOnly is available)",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceVdcVdcAccessControlCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	var accessControl types.ControlAccessParams
+
+	isSharedWithEveryone := d.Get("shared_with_everyone").(bool)
+	everyoneAccessLevel := d.Get("everyone_access_level").(string)
+	sharedList := d.Get("shared_with").(*schema.Set).List()
+
+	// Do some checks before proceeding to contact the API
+	if isSharedWithEveryone && len(sharedList) > 0 {
+		return diag.Errorf("if shared_with_everyone is set to true, shared_with must be an empty set")
+	}
+
+	if isSharedWithEveryone {
+		accessControl.IsSharedToEveryone = true
+		accessControl.EveryoneAccessLevel = takeStringPointer(everyoneAccessLevel)
+	} else {
+		adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+		if err != nil {
+			return diag.Errorf("error when retrieving AdminOrg - %s", err)
+		}
+
+		AccessSettings, err := sharedSetToAccessControl(adminOrg, sharedList)
```

Comment:
why this variable starts with an upper case?

Reply:
Changed, not sure why I did that, probably misclicked shift button